### ISSUE TITLE
Add a manifest and service worker and all that junk

### ISSUE THREAD

**Issue by Rich-Harris** [alizauf/the-bike-electric#1]
...so that we Android users can [add it to the home screen](https://developers.google.com/web/fundamentals/app-install-banners/)

**Comment by alizauf** [comment]
Good call. If I can't find any e-bikes, might work on adding that this weekend.

**Comment by josephfrazier** [comment]
Does anyone know if adding it to the home screen means that it can run in the background, so your phone could be in your pocket with the screen off, but still notify you when there's a bike nearby?

I've been working on a much less polished version of this project at https://reported-web.herokuapp.com/electricitibikes (code at https://github.com/josephfrazier/Reported-Web/blob/51f31cae45f5f554252a102a50744d36e8879c33/src/routes/electricitibikes/ElectriCitibikes.js), and getting it to work without being in the foreground was a goal of mine, but I don't know how easy that is.

EDIT: Getting the current position might also not be possible, which could pose a problem for the use case of: I'm riding a manual citibike, but would like to switch to an electric one if possible https://github.com/w3c/ServiceWorker/issues/745

**Comment by Rich-Harris** [comment]
Background notifications are somewhat involved — basically there needs to be a server that maintains a list of clients (service workers) and can send notifications to them, and the service worker needs to have some code like

```js
self.addEventListener('push', event => {
  // some magic happens
});
```

There's a [guide here](https://developers.google.com/web/fundamentals/push-notifications/) — the length of it tells you how much of a nuisance it is to set up. It's definitely doable, it's just a lot more work than foreground notifications.

**Comment by josephfrazier** [comment]
Thanks for the info! That's what I was suspecting, that it couldn't be completely client-side anymore, but I was still a bit unsure. Maybe I'll get around to it someday... :P

**Comment by Rich-Harris** [comment]
Unless the service worker was responsible for polling the stations you've marked, perhaps... I don't know if that's possible but it might be. Hmm, now I want to investigate 🤔

**Comment by josephfrazier** [comment]
Yeah, I'm currently using [Server-sent events](https://en.wikipedia.org/wiki/Server-sent_events), but I don't know if those keep going to a backgrounded PWA.

**Comment by RichardMaher** [comment]
ServiceWorkers need to simply implement the TravelManager detailed [here](https://github.com/RichardMaher/Brotkrumen)
